Update color limits for _wrapped_collection_fix automatically fixing #1654

LiamBindle opened the pull request:
Hi everyone. This is my first PR to cartopy but I've been interested in doing so for a while. I apologize if I've followed any guidelines incorrectly.  

This PR fixes #1654. It adds an override to GeoQuadMesh such that if the `_wrapped_collection_fix` attribute exists `set_clim()` also gets called on `self._wrapped_collection_fix`.

## Rationale

<!-- Please provide detail as to *why* you are making this change. -->
After #1622 cells that cross the antimeridian are a part of a different collection (`_wrapped_collection_fix`). Therefore, to update color limits on those cells, `set_clim()` must be explicitly called.

## Implications

I'm not aware of any, however, if the `_wrapped_collection_fix` collection is removed, so should these changes.


Let me know if you have any questions, or if you'd like any changes.

LiamBindle commented:
Oops, missed the CLA. I signed it

greglucas commented:
Thanks, @LiamBindle, this looks great! My only request is to add a test to make sure your `set_clim()` got dispatched properly. Something like `get_clim() == expected` on both of the collections.

LiamBindle commented:
@greglucas I just added some tests. I implemented a pcolormesh_global_wrap3 image test where the initial norm is incorrect. The test only passes if `set_clim()` operates on both collections. Thanks for your guidance. Let me know if there's anything else.

dopplershift commented on lib/cartopy/mpl/geocollection.py:
```diff
@@ -36,3 +36,11 @@ def set_array(self, A):
+            self._wrapped_collection_fix.set_clim(vmin, vmax)
+
+        # Update color limits for the rest of the cells.
+        super(QuadMesh, self).set_clim(vmin, vmax)
```
Since we're only on Python 3 now this can just be:
```python
super().set_clim(vmin, vmax)
``

LiamBindle replied:
THanks for the suggestion @dopplershift. I just pushed that update.

greglucas reviewed (approved):
Looks good to me. Thanks, @LiamBindle, hope to have more PRs from you in the future!
